cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/TableUpdater.java

```diff
@@ -94,10 +101,10 @@ public void run() {
           }
           HashMap<String, Long> updatesToWrite;
           synchronized (cachedUpdatesLock) {
-            // Make a copy so HBase write can happen outside of the synchronized block
-            updatesToWrite = cachedUpdates;
-            // Remove stale updates
-            cachedUpdates = new HashMap<>();
+            // Make a copy of new updates so HBase write can happen outside of the synchronized block
+            updatesToWrite = getNewEntries(cachedUpdates);
+            // Mark entries stale
+            markMapDirty(cachedUpdates);
```

Comment:
you can do this when you're making the copy to save an iteration through the map